### ISSUE TITLE
Lurker infinite loop on using directory as resource

### ISSUE THREAD

**Issue by vpassapera** [flint/Lurker#27]
```php
        $watcher = new \Lurker\ResourceWatcher();
        $watcher->track('provision', __DIR__ . '/.provision');

        $watcher->addListener('provision', function (\Lurker\Event\FilesystemEvent $e) {
            $this->say('----------------------------------------------------------------------');
            $this->say('A change was detected in the provisioning directory...');
            $this->say('Vagrant reprovision might be required based on changes.');
            $this->say('----------------------------------------------------------------------');
        });
```
```
...
PHP 224. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 225. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 226. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 227. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 228. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 229. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 230. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 231. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 232. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 233. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 234. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 235. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 236. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 237. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 238. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 239. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 240. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 241. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 242. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 243. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 244. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 245. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 246. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 247. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 248. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 249. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 250. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 251. Lurker\StateChecker\DirectoryStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:97
PHP 252. Lurker\StateChecker\DirectoryStateChecker->createDirectoryChildCheckers() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/DirectoryStateChecker.php:22
PHP 253. Lurker\StateChecker\FileStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/NewDirectoryStateChecker.php:99
PHP 254. Lurker\StateChecker\FileStateChecker->__construct() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/FileStateChecker.php:23
PHP 255. Lurker\Resource\FileResource->getModificationTime() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/StateChecker/ResourceStateChecker.php:29
PHP 256. Lurker\Resource\FileResource->exists() /vagrant/vendor/henrikbjorn/lurker/src/Lurker/Resource/FileResource.php:14

```

**Comment by vpassapera** [comment]
Realized that i had to limit the number of items.
